assets/eip-7265/src/utils/LimiterLib.sol

```diff
@@ -0,0 +1,136 @@
+        limiter.minLiqRetainedBps = minLiqRetainedBps;
+        limiter.limitBeginThreshold = limitBeginThreshold;
+    }
+
+    function recordChange(Limiter storage limiter, int256 amount, uint256 withdrawalPeriod, uint256 tickLength)
+        internal
+    {
+        // If token does not have a rate limit, do nothing
+        if (!initialized(limiter)) {
+            return;
+        }
+
+        uint256 currentTickTimestamp = getTickTimestamp(block.timestamp, tickLength);
+        limiter.liqInPeriod += amount;
+
+        uint256 listHead = limiter.listHead;
+        if (listHead == 0) {
+            // if there is no head, set the head to the new inflow
+            limiter.listHead = currentTickTimestamp;
+            limiter.listTail = currentTickTimestamp;
+            limiter.listNodes[currentTickTimestamp] = LiqChangeNode({amount: amount, nextTimestamp: 0});
+        } else {
+            // if there is a head, check if the new inflow is within the period
+            // if it is, add it to the head
+            // if it is not, add it to the tail
+            if (block.timestamp - listHead >= withdrawalPeriod) {
+                sync(limiter, withdrawalPeriod);
+            }
+
+            // check if tail is the same as block.timestamp (multiple txs in same block)
+            uint256 listTail = limiter.listTail;
+            if (listTail == currentTickTimestamp) {
+                // add amount
+                limiter.listNodes[currentTickTimestamp].amount += amount;
+            } else {
+                // add to tail
+                limiter.listNodes[listTail].nextTimestamp = currentTickTimestamp;
+                limiter.listNodes[currentTickTimestamp] = LiqChangeNode({amount: amount, nextTimestamp: 0});
+                limiter.listTail = currentTickTimestamp;
+            }
+        }
+    }
+
+    function sync(Limiter storage limiter, uint256 withdrawalPeriod) internal {
+        sync(limiter, withdrawalPeriod, type(uint256).max);
+    }
+
+    function sync(Limiter storage limiter, uint256 withdrawalPeriod, uint256 totalIters) internal {
+        uint256 currentHead = limiter.listHead;
+        int256 totalChange = 0;
+        uint256 iter = 0;
+
+        while (currentHead != 0 && block.timestamp - currentHead >= withdrawalPeriod && iter < totalIters) {
+            LiqChangeNode storage node = limiter.listNodes[currentHead];
+            totalChange += node.amount;
+            uint256 nextTimestamp = node.nextTimestamp;
+            // Clear data
+            limiter.listNodes[currentHead];
+            currentHead = nextTimestamp;
+            // forgefmt: disable-next-item
+            unchecked {
+                ++iter;
+            }
+        }
+
+        if (currentHead == 0) {
+            // If the list is empty, set the tail and head to current times
+            limiter.listHead = block.timestamp;
+            limiter.listTail = block.timestamp;
```

Comment:
Why `block.timestamp` is used here and not `currentTickTimestamp`? It would be cleaner that way.